### ISSUE TITLE
Investigate what infrastructure do we need in order to have the ability of shipping a subset of packages on a release

### ISSUE THREAD

**Issue by joperezr** [dotnet/aspire#4328]
Today, in order to be safe, when we want to ship a change we need to ship the whole set of packages. This is because we don't have a good way of knowing exactly what is the full subset that needs to be shipped given a specific change or list of packages that we want to service. This means that if we tried to ship a subset of packages today, there is a risk that we may miss shipping some packages that are transitive dependencies of the packages that we are shipping, templates that need updating, or packages that are referenced by new templates that are being shipped.

This issue is to track the investigation of what it would take to have good confidence that, given a change that we want to ship and set of packages that we want to service, we'd know the full set of packages that we need to ship from the build, and which ones is okay to not ship. Being able to do this confidently, will increase our agility.

cc: @DamianEdwards, @RussKie, @radical

**Comment by mitchdenny** [comment]
@joperezr I'm moving this to the backlog. If it still ranks up there for priority move it to 8.2?

**Comment by joperezr** [comment]
Yeah, I think that is fair. TBH this is a process that might be a bit harder than expected, as it is also something that no team has fixed yet, so I think at least for now it is something we don't need as a P0 thing

**Comment by mitchdenny** [comment]
Azure SDK does this ... it's hard work (speaking from experience).

**Comment by davidfowl** [comment]
@joperezr still relevant?

**Comment by joperezr** [comment]
We haven't been able to prioritize this and it would at somepoint be nice as I'm sure there will be many cases where we want to only ship a subset of the packages. I'd probably keep it open but just put it at low priority, mostly because I do hope that we get to this eventually.
